lib/executor/ramping_arrival_rate.go

```diff
@@ -316,25 +316,37 @@ func (varr RampingArrivalRate) Run(parentCtx context.Context, out chan<- stats.S
 	returnedVUs := make(chan struct{})
 	startTime, maxDurationCtx, regDurationCtx, cancel := getDurationContexts(parentCtx, duration, gracefulStop)
 
+	iterations := make(chan lib.ActiveVU)
+
 	defer func() {
 		// Make sure all VUs aren't executing iterations anymore, for the cancel()
 		// below to deactivate them.
 		<-returnedVUs
 		cancel()
+		close(iterations)
 		activeVUsWg.Wait()
 	}()
+
 	activeVUs := make(chan lib.ActiveVU, maxVUs)
 	activeVUsCount := uint64(0)
 
 	returnVU := func(u lib.InitializedVU) {
 		varr.executionState.ReturnVU(u, true)
 		activeVUsWg.Done()
 	}
+
+	runIterationBasic := getIterationRunner(varr.executionState, varr.logger)
 	activateVU := func(initVU lib.InitializedVU) lib.ActiveVU {
 		activeVUsWg.Add(1)
 		activeVU := initVU.Activate(getVUActivationParams(maxDurationCtx, varr.config.BaseConfig, returnVU))
 		varr.executionState.ModCurrentlyActiveVUsCount(+1)
 		atomic.AddUint64(&activeVUsCount, 1)
+		go func() {
+			for avu := range iterations {
```

Comment:
Hmm this seems a bit strange :confused: 

I haven't tested it and I haven't spotted an actual bug, but sending a VU (i.e. the _worker_) over the `iterations` (i.e. the _work_) channel is counter-intuitive. Especially when you spawn this goroutine in the context of `Activate()`-ing a specific VU, but the `avu` variable here would be assigned any random VU that is free when running the iteration.